app/model/onnx_runtime.py

```
@@ -0,0 +1,225 @@
+
+        # Minimal tokenizer fallback: single token spanning entire text
+        offsets: List[Tuple[int, int]] = [(0, len(text))]
+        prepared_inputs: Dict[str, Any] = {"text": text, "offsets": offsets}
+        # store input text for mapping stage
+        self._last_input_text = text
+
+        try:
+            task = asyncio.to_thread(self._run_inference_sync, prepared_inputs)
+            raw = await asyncio.wait_for(task, timeout=timeout_ms / 1000.0)
+        except asyncio.TimeoutError as e:
+            raise InferenceTimeout(f"Inference exceeded timeout of {timeout_ms}ms") from e
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)